posthog/api/test/test_survey.py

```diff
@@ -450,6 +450,9 @@ def test_updating_survey_to_remove_targeting_deletes_targeting_flag(self):
         assert updated_survey_deletes_targeting_flag.json()["name"] == "survey with targeting"
         assert updated_survey_deletes_targeting_flag.json()["targeting_flag"] is None
 
+        flags = self.client.get(f"/api/projects/{self.team.id}/feature_flags/").json()
```

Comment:
Here was the issue.

1. You create a survey with a targeting flag and save
2. You update the survey and remove the targeting flag and save
3. You decide to re-add a targeting condition on the survey, but will see an error `this flag already exists` even though there's nothing associated with the survey anymore
4. You go to `feature flags` and see the original targeting flag showing up there 

Now, in the update call for `survey.py` the targeting flag is actually deleted and we disassociate the targeting flag ID with the survey. However, directly after that we check for `instance.targeting_flag` which will return true even though we just deleted our flag, and we call `.save()` which seems to re-create the flag.

Checking FeatureFlag.objects will say the flag is deleted, which is true, but then the flag gets re-created sometime in between operations and we're not able to detect it until we call get feature flags.

Does that make more sense?

I could also check to make sure `remove_targeting_flag` doesn't exist in the filters before saving the flag, which would make it clearer, but a separate test would just be a duplicate of this one no?

Reply:
I see. The problem is this test isn't testing what you want - it passes with or without your change - have updated it to do the right thing - the flagId changes, which is why the old test didn't catch it 😮‍💨

Reply:
wait that shouldn't be true though because it kept failing without the change 🤔

Reply:
hmm, not sure if I'm doing something different, but the test passed for me with & without the change. Which makes sense to me, since your check and the check right below, for FeatureFlag.objects.get() is equivalent